nuxeo-core/nuxeo-core-schema/src/main/java/org/nuxeo/ecm/core/schema/SchemaManagerImpl.java

```diff
@@ -237,6 +237,22 @@ public DocumentTypeDescriptor getDocumentTypeDescriptor(String name) {
         return last;
     }
 
+    // NXP-14218: used for tests, to be able to unregister it
+    public FacetDescriptor getFacetDescriptor(String name) {
+        FacetDescriptor desc = allFacets.stream().filter(f -> f.name.equals(name)).reduce((a, b) -> b).get();
+        return desc;
+    }
+
+    // NXP-14218: used for tests, to recompute available facets
+    public void recomputeDynamicFacets() {
+        recomputeFacets();
```

Comment:
Why not make `recomputeFacets()` public and add a comment about visibility and its use in tests ?